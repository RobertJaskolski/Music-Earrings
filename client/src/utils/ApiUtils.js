import { tokensActions } from "../reducers/tokens";
import { authActions } from "../reducers/auth";

export const options = (Addurl, token, method = "GET") => ({
  method: method,
  url: `https://api.spotify.com${Addurl}`,
  timeout: 2000,
  headers: {
    Accept: "application/json",
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  },
});
export const optionsPOST = (Addurl, token, requestBody, method = "POST") => ({
  method: method,
  url: `https://api.spotify.com${Addurl}`,
  timeout: 2000,
  data: requestBody,
  headers: {
    Accept: "application/json",
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  },
});

export const optionsPUT = (Addurl, token, requestBody, method = "PUT") => ({
  method,
  url: `https://api.spotify.com${Addurl}`,
  timeout: 2000,
  data: requestBody,
  headers: {
    Accept: "application/json",
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  },
});

export const LogoutUser = (dispatch) => {
  dispatch(authActions.logout());
  dispatch(tokensActions.clear());
};
export const searchValue = (searchText = "") => {
  if (searchText) {
    return searchText;
  }
  return undefined;
};
