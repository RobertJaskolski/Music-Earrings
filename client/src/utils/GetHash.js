import queryString from "query-string";

const GetHash = (refresh) => {
  if (window.location.hash) {
    const tokens = queryString.parse(window.location.hash);
    refresh({
      accessToken: tokens["access_token"],
      refreshToken: tokens["refresh_token"],
    });
    window.history.replaceState(null, null, " ");
  }
};

export default GetHash;
