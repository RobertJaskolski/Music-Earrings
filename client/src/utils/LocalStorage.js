export const checkTokens = () => {
  try {
    const serializedTokens = localStorage.getItem("tokens");
    if (serializedTokens === null) {
      return undefined;
    }
    return JSON.parse(serializedTokens);
  } catch (err) {
    return undefined;
  }
};

export const saveTokens = (tokens) => {
  try {
    const serializedTokens = JSON.stringify(tokens);
    localStorage.setItem("tokens", serializedTokens);
  } catch (err) {
    console.log(err);
  }
};
