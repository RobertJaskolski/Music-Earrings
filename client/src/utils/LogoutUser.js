import { tokensActions } from "../reducers/tokens";
import { authActions } from "../reducers/auth";

const LogoutUser = (dispatch) => {
  dispatch(authActions.logout());
  dispatch(tokensActions.clear());
};

export default LogoutUser;
