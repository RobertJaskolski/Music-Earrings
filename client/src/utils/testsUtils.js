import checkPropTypes from "check-prop-types";
export { default as testStore } from "../store";
export const findByDataTest = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
  return propsErr;
};
