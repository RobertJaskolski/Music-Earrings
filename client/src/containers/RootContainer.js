import React, { useEffect } from "react";
import { Container, Grid } from "@material-ui/core";
import Nav from "./Nav/Nav";
import Footer from "../components/Footer/Footer";
import { connect } from "react-redux";
import { tokensActions } from "../reducers/tokens";
import MyAPI from "../api/MyAPI";
import GetHash from "../utils/GetHash";
import withAuthorizedState from "../components/shared/HOC/withAuthorized";
import withUserProfileState from "../components/shared/HOC/withUserProfile";
import API from "../api/SpotifyAPI";
import { compose } from "recompose";
import SearchResults from "./SearchResults/SearchResults";
import FavArtists from "./FavArtists/FavArtists";
import Tracklist from "./Tracklist/Tracklist";

const WithAuthorizedAndUserInfoFavArtists = compose(
  withAuthorizedState,
  withUserProfileState
)(FavArtists);
const WithAuthorizedAndUserInfoNav = compose(
  withAuthorizedState,
  withUserProfileState
)(Nav);
const WithAuthorizedSearchResults = withAuthorizedState(SearchResults);
const WithAuthorizedTracklist = withAuthorizedState(Tracklist);
function RootContainer({
  refresh,
  getUserProfile,
  auth,
  refreshToken,
  getFavArtists,
  getRefreshToken,
}) {
  useEffect(() => {
    GetHash(refresh);
    getRefreshToken(refreshToken);
    if (auth) {
      getUserProfile();
      getFavArtists();
    }
  });
  return (
    <Container maxWidth='xl'>
      <Grid item xs={12}>
        <WithAuthorizedAndUserInfoNav />
      </Grid>
      <Grid item xs={12}>
        <WithAuthorizedSearchResults data-test='searchResults' />
      </Grid>
      <Grid item xs={12}>
        <WithAuthorizedAndUserInfoFavArtists />
      </Grid>
      <Grid item md={8} xs={12}>
        <WithAuthorizedTracklist />
      </Grid>
      <Grid item xs={12}>
        <Footer />
      </Grid>
    </Container>
  );
}

const mapDispatchToProps = (dispatch) => ({
  refresh: (tokens) => dispatch(tokensActions.refresh(tokens)),
  getUserProfile: () => dispatch(API.GetUserProfile()),
  getFavArtists: () => dispatch(API.GetUserFavArtists()),
  getRefreshToken: (refreshToken) => dispatch(MyAPI.RefreshToken(refreshToken)),
});
const mapStateToProps = (state) => {
  return {
    refreshToken: state.tokens["refreshToken"],
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(RootContainer);
