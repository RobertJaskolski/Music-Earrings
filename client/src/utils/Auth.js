import { loadTokens } from "./LocalStorage";
import axios from "axios";
import store from "../index";
import { tokensActions } from "../reducers/tokens";
import { authActions } from "../reducers/auth";

const checkAuth = () => {
  if (store.getState().tokens["refreshToken"]) {
    axios
      .get(
        `${process.env.REACT_APP_API_URL}/refresh_token?refresh_token=${
          store.getState().tokens["refreshToken"]
        }`
      )
      .then((res) => {
        tokensActions.refresh({
          accessToken: res.data["access_token"],
          refreshToken: store.getState().tokens["refreshToken"],
        });
        store.dispatch(authActions.login());
      })
      .catch((err) => {
        store.dispatch(authActions.logout());
        store.dispatch(tokensActions.clear());
      });
  }
};

export default checkAuth;
