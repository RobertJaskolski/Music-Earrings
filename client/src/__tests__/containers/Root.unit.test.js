import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RootContainer from '../../containers/RootContainer';
import { testStore } from '../../utils/testsUtils';

const makeStore = (INIT_STATE = {}) => {
  const store = testStore(INIT_STATE);
  return store;
};

describe('Root container', () => {
  test('should render component', () => {
    const store = makeStore();
    const wrapper = shallow(<RootContainer store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
