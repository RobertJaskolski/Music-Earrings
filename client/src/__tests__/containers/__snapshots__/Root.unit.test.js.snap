// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root container should render component 1`] = `
<ContextProvider
  value={null}
>
  <RootContainer
    getFavArtists={[Function]}
    getRefreshToken={[Function]}
    getUserProfile={[Function]}
    refresh={[Function]}
    refreshToken=""
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
